src/shared/dissect-image.h

```diff
@@ -59,6 +61,7 @@ typedef enum DissectImageFlags {
         DISSECT_IMAGE_MOUNT_NON_ROOT_ONLY = 1 << 7,  /* Mount only non-root partitions */
         DISSECT_IMAGE_VALIDATE_OS         = 1 << 8,  /* Refuse mounting images that aren't identifiable as OS images */
         DISSECT_IMAGE_NO_UDEV             = 1 << 9,  /* Don't wait for udev initializing things */
+        DISSECT_IMAGE_RELAX_VAR_CHECK     = 1 << 10, /* Don't insist that the UUID of /var is hashed from /etc/machine-id */
```

Comment:
I'd turn this around, i.e. add DISSECT_IMAGE_STRICT_VAR_CHECK. It seem that only gpt-auto-generator would use that switch, and most tools would do the relaxed check.

I'm also not convinced if such a strict check is necessary. But that is OK: we can always relax the check later if it turns out that there's legitimate use cases.

Reply:
I am mostly concerned about having two distros on the same hdd installed and they fight for each other's partition. We can get away with that for /var/tmp and mostly for /srv and /home, but really not for /var, so I figured we better stay away from that and are extra careful.

And I figured i make the check opt-out rather than opt-in also to be extra careful we are not too agressive with mount stuff that isn't ours.

i.e. both times it's about being extra careful with automatisms....

but anyway, we can certainly discuss this independently of this specific PR.